### ISSUE TITLE
Ambiguous timezone info in datetime

### ISSUE THREAD

**Issue by eumiro** [mverleg/pyjson_tricks#89]
Storing datetime objects using year/month/day/hour/minute/second and the name of the timezone (e.g. `Europe/Paris`) is unfortunately not enough during the end of daylight saving time.

Both 2023-10-29 00:30 and 01:30 UTC map to 02:30 of Paris time, the former in DST (CEST), while the latter in standard time (CET), which is but not stored in the JSON version. The DST switch happens at 01:00 UTC when the local clocks move from 03:00 CEST to 02:00 CET.

The following code:

```
import datetime
import pytz
from json_tricks import dumps

print(dumps(datetime.datetime(2023, 10, 29, 0, 30, 0, 0, datetime.UTC).astimezone(pytz.timezone("Europe/Paris"))))
print(dumps(datetime.datetime(2023, 10, 29, 1, 30, 0, 0, datetime.UTC).astimezone(pytz.timezone("Europe/Paris"))))
```

prints

```
{"__datetime__": null, "year": 2023, "month": 10, "day": 29, "hour": 2, "minute": 30, "tzinfo": "Europe/Paris"}
{"__datetime__": null, "year": 2023, "month": 10, "day": 29, "hour": 2, "minute": 30, "tzinfo": "Europe/Paris"}
```

When I `loads` any of these JSON objects, it becomes `2023-10-29 02:30:00+01:00`, which is correct for `01:30 UTC`, but one hour off `00:30 UTC`.

There should be some sort of a `fold` attribute to distinguish the two cases.

**Comment by mverleg** [comment]
Nice catch thanks! I can't get `fold` to work but I have a proposed fix in #93, dwyt?

**Comment by mverleg** [comment]
Merged in v3.17.3, reopen if any issues
